### PR TITLE
UCT/IB: pass error status thru handle_failure and set_ep_failed

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
Fixes IB part of https://github.com/openucx/ucx/issues/1956

### CODE REVIEW

File: src/uct/ib/rc/verbs/rc_verbs_common.h [hunk 1]
```diff
@@ -142,6 +142,8 @@ extern ucs_config_field_t uct_rc_verbs_iface_common_config_table[];
 
 void uct_rc_verbs_txcnt_init(uct_rc_verbs_txcnt_t *txcnt);
 
+ucs_status_t uct_rc_verbs_wc2ucs_status(enum ibv_wc_status status);
```
Comment (alinask) [inline src/uct/ib/rc/verbs/rc_verbs_common.h]: _wc_to_ ...

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5095/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5096/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5097/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5098/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5104/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5111/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5116/ for details (Mellanox internal link).

**Pull request merged**